File: cookbook/java/createActions/createSmsCampaignAction.java
package createActions;

import java.io.FileInputStream;
import java.io.IOException;

import java.util.Map;
import java.util.Properties;

import Utils.Request;
import org.json.JSONException;
import org.json.JSONObject;

public class createSmsCampaignAction
{
	public static void main(String[] args) throws IOException, JSONException
	{
        Properties properties = new Properties();
        try {
            properties.load(new FileInputStream("config.properties"));
        } catch (IOException e) {
            System.out.print(e.getMessage());
        }
        String xKey = properties.getProperty("xKey");
        String baseUrl = properties.getProperty("url");

        String type = "smsCampaign";	//Code pour envoyer une campagne de SMS
        String name = "SMSCampaignFromApi (java)";	//Nom de l'action
        String description = "SMSCampaignFromApi (java)";	//Description de l'action

        Integer informationFolder = 0123;	//Id du dossier dans lequel vous voulez mettre l'action ('null' pour aucun dossier)
        Integer informationCategory = 0123;	//Id de la categorie de campagne (Infos compte > Parametrage > Categories de campagnes)

        String textContent = "Text message";	//Message texte

        //On trouve l'adresse pour la requete
        String url = baseUrl + "actions";

        //Creation du Json du message
        JSONObject informations = new JSONObject();
        informations.put("folder", informationFolder);
        informations.put("category", informationCategory);

        JSONObject content = new JSONObject();
        content.put("textContent", textContent);

        JSONObject scheduler = new JSONObject();
        scheduler.put("type", "scheduled");	//Envoie : immediat = 'asap' / Date = 'scheduled'
        //scheduler.put("startDate", "2015-07-27T08:15:00Z");	//Si type = 'scheduled' sinon a enlever

        JSONObject jsonMessage = new JSONObject();
        jsonMessage.put("type", type);
        jsonMessage.put("name", name);
        jsonMessage.put("description", description);
        jsonMessage.put("informations", informations);
        jsonMessage.put("content", content);
        jsonMessage.put("scheduler", scheduler);

        //On affiche le message
        System.out.print(jsonMessage + "\n");

        //On affiche le message
        System.out.print(jsonMessage + "\n");

        Map<String, String> resp = Request.connection(url, xKey, jsonMessage, "POST");

        if (resp.get("mess").equals("OK")) {
            //Lecture des donnees
            System.out.print("Action : " + name + " created.\n\n"
                    + "Don\'t forget to normalize your phone numbers with your country.");
        } else {
            System.out.print("Error " + resp.get("code") + ": " + resp.get("mess") + "\n");
        }
	}
}

File: cookbook/php/createActions/createSmsCampaignAction.php
<?php

//Ici, renseignez la xKey
$xKey = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';

$type = 'smsCampaign';	//Code pour envoyer une campagne de SMS
$name = 'SMSCampaignFromApi (php)';	//Nom de l'action
$description = 'SMSCampaignFromApi (php)';	//Description de l'action

$informationFolder = 0123;	//Id du dossier dans lequel vous voulez mettre l'action ('null' pour aucun dossier)
$informationCategory = 0123;	//Id de la categorie de campagne (Infos compte > Parametrage > Categories de campagnes)

$textContent = 'Text message';	//Texte du message

$idTestSegment = 0123;	//Id du segment de test pour la validation


//On trouve l'adresse pour la requete
$url = 'http://v8.mailperformance.com/actions';

//Creation du Json du message
$arr = array(
    'type' => $type,
    'name' => $name,
    'description' => $description,
    'informations' => array(
        'folder' => $informationFolder,
        'category' => $informationCategory),
    'scheduler' => array(
        'type' => 'asap',	//Envoie : immediat = 'asap' / Date = 'scheduled'
        //'startDate' => '2016-07-27T08:15:00Z',	//Si type = 'scheduled' sinon a enlever)
        )
    'content' => array(
        'textContent' => $textContent));


//On affiche le message
$message = json_encode($arr);
echo $message . "\n";

//Connexion
$con = connect($url, $xKey, $message);

$result = $con['result'];
$info = $con['info'];
$req = $con['req'];

if ($info['http_code'] != 200)
{
    echo 'Error : ' . $info['http_code'];
}
else
{
    //L'action a bien ete creer
    echo 'Action : ' . $name . " created.\n\n";
    echo 'Don\'t forget to normalize your phone numbers with your country.';
}
curl_close($req);



//Fonctions -----



//Utilisation de cURL pour remplir les requetes
function startCurlInit($url)
{
    $init = curl_init();
    curl_setopt($init, CURLOPT_URL, $url);
    curl_setopt($init, CURLOPT_RETURNTRANSFER, true);
    return ($init);
}

//Fonction de connexion
function connect($url, $xKey, $message)
{
    //On remplit la requete
    $req = startCurlInit($url);
    curl_setopt($req, CURLOPT_CUSTOMREQUEST, 'POST');

    //Mise en place du xKey et des options
    curl_setopt($req, CURLOPT_HTTPHEADER, array(
    'X-Key: ' . $xKey,
    'Content-Type: application/json',
    'Content-Length: ' . strlen($message)));
    curl_setopt($req, CURLOPT_POSTFIELDS, $message);

    //Execution de la requete
    $result = curl_exec($req);

    //Verification des reponses
    $info = curl_getinfo($req);

    return (array('result' => $result, 'info' => $info, 'req' => $req));
}

?>
